test/specs/EventsSpec.js

```diff
@@ -21,325 +21,355 @@ describe("Event Tests", function() {
+        spy = jasmine.createSpy('spy');
       });
 
-      it("'slide' event is triggered properly and can be binded to", function() {
-        testSlider.on('slide', function() {
-          flag = true;
+      it("'slideStart' event is triggered properly and can be binded to", function(done) {
+        testSlider.on('slideStart', spy);
+        testSlider.data('slider')._mousedown(mouse);
+        setTimeout(function() {
+          expect(spy).toHaveBeenCalled();
+          done();
         });
+      });
+
+      it("'slide' event is triggered properly and can be binded to", function(done) {
+        testSlider.on('slide', spy);
         testSlider.data('slider')._mousemove(mouse);
-        expect(flag).toBeTruthy();
+        setTimeout(function() {
+          expect(spy).toHaveBeenCalled();
+          done();
+        });
       });
 
-      it("'slide' event sets the right value on the input", function() {
+      it("'slide' event sets the right value on the input", function(done) {
         testSlider.on('slide', function() {
-          flag = true;
           expect(isNaN(testSlider.val())).not.toBeTruthy();
+          done();
         });
         testSlider.data('slider')._mousemove(mouse);
-        expect(flag).toBeTruthy();
       });
 
-      it("'slide' event value and input value properties are synchronous", function() {
+      it("'slide' event value and input value properties are synchronous", function(done) {
         testSlider.on('slide', function(e) {
-          flag = true;
           expect(e.value.toString()).toEqual(this.value);
+          done();
         });
         testSlider.slider("setValue", 3, true, false);
-
-        expect(flag).toBeTruthy();
       });
 
-      it("'change' event value and input value properties are synchronous", function() {
+      it("'change' event value and input value properties are synchronous", function(done) {
         testSlider.on('change', function(e) {
-          flag = true;
           expect(e.value.newValue.toString()).toEqual(testSlider.val());
+          done();
         });
         testSlider.slider("setValue", 3, false, true);
-
-        expect(flag).toBeTruthy();
       });
 
-      it("'slideStop' event is triggered properly and can be binded to", function() {
-        testSlider.on('slideStop', function() {
-          flag = true;
-        });
+      it("'slideStop' event is triggered properly and can be binded to", function(done) {
+        testSlider.on('slideStop', spy);
         testSlider.data('slider')._mouseup(mouse);
-        expect(flag).toBeTruthy();
+        setTimeout(function() {
+          expect(spy).toHaveBeenCalled();
+          done();
+        });
       });
 
-      it("slider should not have duplicate events after calling 'refresh'", function() {
-        flag = 0;
-        testSlider.on('slideStop', function() {
-          flag += 1;
-        });
+      it("slider should not have duplicate events after calling 'refresh'", function(done) {
```

Comment:
This test makes sure that `slideStop` handler is not called more than once, hence the `flag` variable.

```js
it("slider should not have duplicate events after calling 'refresh'", function(done) {
    flag = 0;
    var obj = {
        addOne: function() {
        flag++;
        }
    };
    spyOn(obj, 'addOne').and.callThrough();

    testSlider.on('slideStop', obj.addOne);

    testSlider.slider('refresh');
    testSlider.data('slider')._mouseup(mouse);

    window.setTimeout(function() {
        expect(flag).toBe(1);
        expect(obj.addOne.calls.count()).toBe(1);
        done();
    }, 0);
});
```